client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java

```diff
@@ -329,4 +318,65 @@ protected final Span startSpan(final Message message) {
             return Future.failedFuture(t);
         });
     }
+
+    /**
+     * Creates a log entry in the given span with the message id and information about the sender (credits, QOS).
+     * 
+     * @param currentSpan The current span to log to.
+     * @param messageId The message id.
+     */
```

Comment:
can you also document the `NullPointerException` being thrown?

Reply:
I've added that.